docs/mp/oci/03_vault.adoc

```diff
@@ -0,0 +1,235 @@
+oci.vault.cryptographic-endpoint: "<...>"
+----
+
+The OCIDs can be set up and found in OCI under Security tab.
+
+image::oci/ocivault.png[OCI Vault, align="center"]
+
+REST endpoint should be setup as follows:
+
+[source,java]
+----
+@Path("/vault")
+public class VaultResource {
+    private final OciVault vault;
+    private final String vaultOcid;
+    private final String compartmentOcid;
+    private final String encryptionKeyOcid;
+    private final String signatureKeyOcid;
+
+    @Inject
+    VaultResource(@Named("custom") OciVault vault, <1>
+                  @ConfigProperty(name = "app.vault.vault-ocid") <2>
+                          String vaultOcid,
+                  @ConfigProperty(name = "app.vault.compartment-ocid") <3>
+                          String compartmentOcid,
+                  @ConfigProperty(name = "app.vault.encryption-key-ocid") <4>
+                          String encryptionKeyOcid,
+                  @ConfigProperty(name = "app.vault.signature-key-ocid") <5>
+                          String signatureKeyOcid) {
+        this.vault = vault;
+        this.vaultOcid = vaultOcid;
+        this.compartmentOcid = compartmentOcid;
+        this.encryptionKeyOcid = encryptionKeyOcid;
+        this.signatureKeyOcid = signatureKeyOcid;
+    }
+}
+----
+
+<1> `OciVault` support class is configured and injected automatically
+<2> <3> <4> <5> Properties a read from the configuration
+
+== OCI Vault usage
+
+=== Encryption
+To encrypt a text, submitted with a GET request to `/encrypt` endpoint:
+
+[source,java]
+----
+@GET
+@Path("/encrypt/{text}")
+public String encrypt(@PathParam("text") String secret) {
+    return vault.encrypt(Encrypt.Request.builder()
+                                 .keyId(encryptionKeyOcid)
+                                 .data(Base64Value.create(secret)))
+            .cipherText();
+}
+----
+
+=== Decryption
+To decrypt a text, submitted with a GET request to `/decrypt` endpoint:
+
+[source,java]
+----
+@GET
+@Path("/decrypt/{text: .*}")
+public String decrypt(@PathParam("text") String cipherText) {
+    return vault.decrypt(Decrypt.Request.builder()
+                                 .keyId(encryptionKeyOcid)
+                                 .cipherText(cipherText))
+            .decrypted()
+            .toDecodedString();
+}
+----
+
+=== Signature
+To retrieve a signature for submitted text via GET request to `/sign` endpoint:
+
+[source,java]
+----
+@GET
+@Path("/sign/{text}")
+public String sign(@PathParam("text") String dataToSign) {
+    return vault.sign(Sign.Request.builder()
+                              .keyId(signatureKeyOcid)
+                              .algorithm(Sign.Request.ALGORITHM_SHA_224_RSA_PKCS_PSS)
+                              .message(Base64Value.create(dataToSign)))
+            .signature()
+            .toBase64();
+}
+----
+
+=== Verification of a signature
+To verify the correctness of the signature to the submitted text via GET request to `/verify` endpoint:
```

Comment:
To verify the validity of the signature of the submitted text, submit the GET request to the `/verify` endpoint: